index.js

```diff
@@ -6,32 +6,59 @@ const { GitHub, context } = require('@actions/github')
 const sleep = dt => new Promise(resolve => setTimeout(resolve, dt))
 
 const main = async () => {
-  const token = core.getInput('github-token')
+  const token = core.getInput('github_token')
   const number = core.getInput('number')
+  const waitForPullRequestUpdated =
+    core.getInput('waitForPullRequestUpdated') === 'true'
 
   const octokit = new GitHub(token)
 
-  const res = await octokit.pulls.get({
-    ...context.repo,
-    pull_number: number
-  })
-  const oldSha = res.data.head.sha
-
-  await octokit.pulls.updateBranch({
-    ...context.repo,
-    pull_number: number,
-    expected_head_sha: oldSha
-  })
+  let numbers
+  if (number) {
+    numbers = [number]
+  } else {
+    const listRes = await octokit.pulls.list({
+      ...context.repo,
+      state: 'open'
+    })
+    numbers = listRes.data.map(pull => pull.number)
+  }
 
-  while (true) {
+  for (const number of numbers) {
     const res = await octokit.pulls.get({
       ...context.repo,
       pull_number: number
     })
-    if (res.data.head.sha !== oldSha) return
+    const oldSha = res.data.head.sha
+    const prMergeable = res.data.mergeable
+    const prMergeableState = res.data.mergeable_state
+
+    // merge automatically if pull request is outdated and checks have passed
+    if (prMergeable && prMergeableState === 'behind') {
```

Comment:
this is also out of scope, there are different actions for auto merging PRs

Reply:
mergeable_state tells us whether or not a pull request can be updated from its base branch

clean = no update is needed
behind = it's out-of-date and can be updated
dirty = it's out-of-date but can't be updated because of a conflict
unknown = GitHub is still checking it

from here: https://github.com/octokit/octokit.net/issues/1763

Reply:
Oh, but I realise the comment I put in the code is misleading and wrong, it should say update not merge

Reply:
I understand, but I mean there are different github actions for merging a PR once it's green and uptodate with master

Reply:
Makes sense. I've updated the comment to reflect that we're just checking whether the pull request can be updated. Are you happy with this or is something more needed?